tests/model_formsets/tests.py

```diff
@@ -1142,6 +1144,55 @@ def test_inline_formsets_with_wrong_fk_name(self):
         with self.assertRaisesMessage(ValueError, message):
             inlineformset_factory(Author, Book, fields="__all__", fk_name="title")
 
+    def test_inline_formsets_with_prefetch_related(self):
+        """
+        Override BaseInlineFormSet.get_instance_by_queryset_index and
+        BaseInlineFormSet.prepare_queryset (#18597).
+        """
+        author = Author.objects.create(pk=1, name="Charles Baudelaire")
+        Book.objects.create(pk=1, author=author, title="Les Paradis Artificiels")
+        Book.objects.create(pk=2, author=author, title="Les Fleurs du Mal")
+        Book.objects.create(pk=3, author=author, title="Flowers of Evil")
+
+        # Check the default issues one query
+        BookFormSet = inlineformset_factory(Author, Book, fields=["title"])
+
+        with CaptureQueriesContext(connection) as ctx:
+            formset = BookFormSet(instance=author)
+            self.assertEqual(len(formset.forms), 6)
+        self.assertEqual(len(ctx), 1)  # Only 1 query
```

Comment:
You can use `self.assertNumQueries()`:
```suggestion
        with self.assertNumQueries(1):
            formset = BookFormSet(instance=author)
            self.assertEqual(len(formset.forms), 6
```